docs/02.API-REFERENCE.md

```diff
@@ -4973,6 +5001,99 @@ main (void)
 - [jerry_substring_to_char_buffer](#jerry_substring_to_char_buffer)
 
 
+# External context functions
+
+## jerry_create_instance
+
+**Summary**
+
+Creates a JerryScript instance for external context.
+
+**Prototype**
+
+```c
+jerry_instance_t *
+jerry_create_instance (uint32_t heap_size,
+                       jerry_instance_alloc_t alloc,
+                       void *cb_data_p);
+```
+
+- `heap_size` - requested heap size of the JerryScript instance
+- `alloc` - function for allocation
+- `cb_data_p` - user data
+- return value
+  - pointer to the newly created JerryScript instance if success
+  - NULL otherwise.
+
+**Example**
+
+[doctest]: # (test="compile")
+
+```c
+#include <stdlib.h>
+#include <pthread.h>
+
+#include "jerryscript.h"
+#include "jerryscript-port.h"
+
+/* A different Thread Local Storage variable for each jerry instance. */
+__thread jerry_instance_t* tls_instance;
+
+jerry_instance_t *
+jerry_port_get_current_instance (void)
+{
+  /* Returns the instance assigned to the thread. */
+  return tls_instance;
+}
+
+/* Allocate JerryScript heap for each thread. */
+static void *
+instance_alloc_fn (size_t size, void *cb_data)
+{
+  (void) cb_data;
+  return malloc(size);
+}
+
+static void *
+thread_function (void *param)
+{
+  tls_instance = jerry_create_instance (512 * 1024,
+                                        instance_alloc_fn,
+                                        NULL);
+  jerry_init (JERRY_INIT_EMPTY);
+  /* Run the JerryScript instance (e.g.: jerry_parse & jerry_run) */
+  jerry_cleanup ();
+
+  return NULL;
+}
+
+#define NUM_OF_THREADS 8
+
+int
+main (void)
+{
+  pthread_t threads[NUM_OF_THREADS];
+
+  /* Create the threads. */
+  for (int i = 0; i < NUM_OF_THREADS; i++) {
+    pthread_create (&threads[i], NULL, thread_function, (void *) (intptr_t) i);
+  }
+
+  /* Wait for the threads to complete, and release their resources. */
+  for (int i = 0; i < NUM_OF_THREADS; i++) {
+    pthread_join (threads[i], NULL);
+  }
+
+  return 0;
+}
+```
+
+**See also**
+
+- [jerry_instance_t](#jerry_instance_t)
+- [jerry_instance_alloc_t](#jerry_instance_alloc_t)
```

Comment:
I don't think we have an example for cross referencing doc MD pages, but somehow the jerry_port_get_instance function should be mentioned here (even if without a link). Right now no text mentions it, it only appears in the example out of the blue, without any hints why it is needed.

Reply:
I was wondering on this too, but could not find a good solution which works both with MD and the generated `jerryscript.net` page.